src/xo-app/self/dashboard/index.js

```diff
@@ -0,0 +1,317 @@
+import ChartistGraph from 'react-chartist'
+import Icon from 'icon'
+import React, { Component } from 'react'
+import _ from 'messages'
+import forEach from 'lodash/forEach'
+import keyBy from 'lodash/keyBy'
+import map from 'lodash/map'
+import store from 'store'
+import { Panel } from 'react-bootstrap-4/lib'
+import { Row, Col } from 'grid'
+import { createGetObject } from 'selectors'
+
+import {
+  connectStore,
+  formatSize,
+  propTypes
+} from 'utils'
+
+import {
+  subscribeGroups,
+  subscribeUsers,
+  subscribeResourceSets
+} from 'xo'
+
+// ===================================================================
+
+const getObject = createGetObject((_, id) => id)
+
+// ===================================================================
+
+export const resolveResourceSets = resourceSets => (
+  map(resourceSets, resourceSet => {
+    const { objects, ...attrs } = resourceSet
+    const resolvedObjects = {}
+    const resolvedSet = {
+      ...attrs,
+      missing: [],
+      objects: resolvedObjects
+    }
+    const state = store.getState()
+
+    forEach(objects, id => {
+      const object = getObject(state, id)
+
+      // Error, missing resource.
+      if (!object) {
+        resolvedSet.missing.push(id)
+        return
+      }
+
+      const { type } = object
+
+      if (!resolvedObjects[type]) {
+        resolvedObjects[type] = [ object ]
+      } else {
+        resolvedObjects[type].push(object)
+      }
+    })
+
+    return resolvedSet
+  })
+)
+
+// ===================================================================
+
+@propTypes({
+  entities: propTypes.array.isRequired
+})
+export class Entities extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      groups: [],
+      users: []
+    }
+  }
+
+  componentWillMount () {
+    const unsubscribeGroups = subscribeGroups(groups => {
+      this.setState({
+        groups: keyBy(groups, 'id')
+      })
+    })
+    const unsubscribeUsers = subscribeUsers(users => {
+      this.setState({
+        users: keyBy(users, 'id')
+      })
+    })
+
+    this.componentWillUnmount = () => {
+      unsubscribeGroups()
+      unsubscribeUsers()
+    }
+  }
+
+  render () {
+    const { state } = this
+
+    return (
+      <div>
+        {map(this.props.entities, id => {
+          const entity = state.users[id] || state.groups[id]
```

Comment:
IMHO it would be better to detect the type of the entity based on where (`users`/`groups`) you find it.